### PR TITLE
Set up documentation using Sphinx

### LINKED ISSUE #140: look into self-writing documentation
There are a couple different options out there to create documentation directly from function docstrings. We should investigate which of these (readthedocs, sphinx, etc) seems the most lightweight and easy for us to adapt

### PR DESCRIPTION
**Pull request by alex-l-kong**
Addresses and closes #140.

Provide documentation support using Sphinx. To reduce the number of questions we get from users of ark, our goal is to provide a link that allows users to hopefully figure out what is broken on their own before coming to us. This also provides a more sophisticated form of documentation.

### CODE REVIEW

**Comment by alex-l-kong** [comment]
Helpful things to address:

1. Directory structure: I've currently set up Sphinx to not separate source and build. Let me know how you think the project is currently structured and if improvements need to be made.
2. Getting autosectionlabel to work. Currently, this does not work as intended (link to subheadings in the same file, would be extremely helpful in index.rst). Probably something about my conf.py file that's messed up. Or maybe the way I'm using toctree.

Feel free to leave any other comments as you see fit!

**Comment by ngreenwald** [comment]
Yup, looks like that's what our elder siblings do as well: https://github.com/vanvalenlab/deepcell-tf/tree/master/docs

**Comment by alex-l-kong** [comment]
> I don't think we need to commit the files resulting from the build (js, css, and html files). Shouldn't the makefile, rst/md index file(s), and sphinx's config.py be sufficient?

I'll remove the files in _build.

**Comment by ackagel** [comment]
shoulda mentioned this in my earlier comment, but you can add `_build/` to the .gitignore so git doesn't get too excited about all those fancy changes

**Comment by alex-l-kong** [comment]
> shoulda mentioned this in my earlier comment, but you can add `_build/` to the .gitignore so git doesn't get too excited about all those fancy changes

Yeah I forgot to do that. Will appear on the next commit.

File: requirements.txt [hunk 1]
```diff
@@ -1,16 +1,27 @@
+commonmark==0.9.1
+h5py==2.10.0
+#ipykernel==5.1.3
+jupyter>=1.0.0,<2
```
Comment (ngreenwald) [inline requirements.txt:4]: deepcell-tf has a separate requirements.txt for the documentation. Let's do that to declutter the main requirements file

File: docs/_markdown/markdown-sample.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
+An h1 header
+============
+
```
Comment (ngreenwald) [inline docs/_markdown/markdown-sample.md:3]: We don't need this file right?

File: docs/markdown-sample.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
+An h1 header
+============
+
+Paragraphs are separated by a blank line.
+
+2nd paragraph. *Italic*, **bold**, and `monospace`. Itemized lists
+look like:
+
+  * this one
```
Comment (ngreenwald) [inline docs/markdown-sample.md:9]: We don't need this file right?

File: docs/_markdown/markdown-sample.md (hunk 2 above)
Comment (alex-l-kong) [reply docs/_markdown/markdown-sample.md:3]: Nope, it was just used to test the base functionality of Sphinx. I'll remove.

File: docs/_markdown/ark.analysis.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+analysis
+====================
+
+These libraries are like ground zero. They are the base libraries to use for analysis.
```
Comment (ackagel) [inline docs/_markdown/ark.analysis.md:4]: Maybe something more like 'These are the base libraries to use for post-segmentation analysis'?

File: docs/_markdown/ark.analysis.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+   :undoc-members:
+   :show-inheritance:
+
+Module contents
+---------------
+
+.. automodule:: ark.analysis
+   :members:
+   :undoc-members:
+   :show-inheritance:
```
Comment (ackagel) [inline docs/_markdown/ark.analysis.md:36]: Can this be deleted or reformatted?  It shows up below the other subpackages as if it were a subpackage.

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+# -- Options for Intersphinx config -------------------------------------------------
+
+# intersphinx mapping, for when there is a cross-reference that has no matching target
+# in the current documentation
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3.6', None),
+    'numpy': ('https://numpy.org/doc/stable', None),
+    'matplotlib': ('https://matplotlib.org/3.2.1', None)
+}
```
Comment (ackagel) [inline docs/conf.py:140]: I think this also adds links to types in docstrings.  That being said, we should find the standard way to name numpy arrays, pandas dataframs, xarrays, etc so they can be linked correctly.

File: docs/_markdown/ark.utils.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+   :undoc-members:
+   :show-inheritance:
+
+Module contents
+---------------
+
+.. automodule:: ark.utils
+   :members:
+   :undoc-members:
+   :show-inheritance:
```
Comment (ackagel) [inline docs/_markdown/ark.utils.md:60]: see earlier comment

File: docs/_markdown/ark.segmentation.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+   :show-inheritance:
+
+
+Module contents
+---------------
+
+.. automodule:: ark.segmentation
+   :members:
+   :undoc-members:
+   :show-inheritance:
```
Comment (ackagel) [inline docs/_markdown/ark.segmentation.md:29]: see earlier comment

File: docs/_markdown/ark.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+   ark.segmentation
+   ark.utils
+
+Module contents
+---------------
+
+.. automodule:: ark
+   :members:
+   :undoc-members:
+   :show-inheritance:
```
Comment (ackagel) [inline docs/_markdown/ark.md:20]: see earlier comment (maybe?)

File: docs/_markdown/ark.analysis.md (hunk 5 above)
Comment (alex-l-kong) [reply docs/_markdown/ark.analysis.md]: Deleted for the most recent commit.

File: docs/conf.py (hunk 6 above)
Comment (alex-l-kong) [reply docs/conf.py]: We can definitely address this in a future PR.

File: docs/_markdown/ark.analysis.md (hunk 4 above)
Comment (alex-l-kong) [reply docs/_markdown/ark.analysis.md:4]: Yeah, I like that 😀.

**Comment by ngreenwald** [comment]
Totally agree. 
Step 1: Merge this PR. 
Step 2: Adam merges in his PR for general documentation cleanup. 
Step 2.5: Alex, you can get started on any non-syntax related remaining issues here while Adam is finishing his PR
Step 3. Once Adam's PR is merged in, clean up any remaining formatting issues. 
Step 4: content review for documentation

**Pull request merged**